examples/MQ/pixelDetector/src/devices/FairMQRunDevice.cxx

```diff
@@ -45,30 +49,61 @@ void FairMQRunDevice::SendObject(TObject* obj, const std::string& chan) {
+            LOG(debug) << "              branch >" << ObjStr->GetString().Data() << "<";
+            std::string modifiedBranchName = std::string("#") + ObjStr->GetString().Data() + "#";
+            if ( mi.first.find(modifiedBranchName) != std::string::npos || mi.first.find("#all#") != std::string::npos ) {
+                if ( (static_cast<FairOnlineSink*>(FairRootManager::Instance()->GetSink()))->IsPersistentBranchAny(ObjStr->GetString()) ) {
+                    LOG(debug) << "Branch \"" << ObjStr->GetString() << "\" is persistent ANY";
+                    if ( ObjStr->GetString().CompareTo("MCTrack") == 0 ) {
+                        TClonesArray** mcTrackArray = (static_cast<FairOnlineSink*>(FairRootManager::Instance()->GetSink()))->GetPersistentBranchAny<TClonesArray**>(ObjStr->GetString());
+                        if ( mcTrackArray ) {
+                            (*mcTrackArray)->SetName("MCTrack");
+                            LOG(debug) << "[" << FairRootManager::Instance()->GetInstanceId() << "] mcTrack " << mcTrackArray << " /// *mcTrackArray " << *mcTrackArray << " /// *mcTrackArray->GetName() " << (*mcTrackArray)->GetName();
+                            TObject* objClone = (*mcTrackArray)->Clone();
+                            LOG(debug) << "FairMQRunDevice::SendBranches() the track array has " << ((TClonesArray*)(objClone))->GetEntries() << " entries.";
+                            FairMQMessagePtr mess(NewMessage());
+                            Serialize<RootSerializer>(*mess,objClone);
+                            parts.AddPart(std::move(mess));
+                            LOG(debug) << "channel >" << mi.first.data() << "< --> >" << ObjStr->GetString().Data() << "<";
+                        }
+                    }
+                    else {
+                        LOG(warning) << "FairMQRunDevice::SendBranches() hasn't got knowledge how to send any branch \"" << ObjStr->GetString().Data() << "\"";
+                        continue;
+                    }
+                }
+                else {
+                    TObject* object   = FairRootManager::Instance()->GetObject(ObjStr->GetString());
+                    if ( object ) {
+                        TObject* objClone = object->Clone();
+                        FairMQMessagePtr mess(NewMessage());
+                        Serialize<RootSerializer>(*mess,objClone);
+                        parts.AddPart(std::move(mess));
+                        LOG(debug) << "channel >" << mi.first.data() << "< --> >" << ObjStr->GetString().Data() << "<";
+                    }
+                    else {
+                        LOG(fatal) << "Object " << ObjStr->GetString() << " NOT FOUND!!!";
+                    }
+                }
+            }
+        }
+        if ( parts.Size() > 0 ) {
+            std::unique_lock<std::mutex> lock(mtx);
+            Send(parts,mi.first.data());
+            mtx.unlock();
```

Comment:
you don't need this line, because the `std::unique_lock` will automatically unlock in it's destructor.